### PR TITLE
Login view refactor

### PR DESCRIPTION
**Pull request by sarthak-marwaha**

### CODE REVIEW

File: src/foam/color/colorlib.js [hunk 1]
```diff
@@ -242,11 +242,12 @@ foam.LIB({
       var scale = desired/gr;
       return [ scale * r, scale * g, scale * b ];
     },
+
     function randomColor( bounds /* { h/s/l: [min, max] } */ ) {
       let hue = bounds?.h ?? [0, 360];
       let sat = bounds?.s ?? [0, 100];
       let lig = bounds?.l ?? [0, 100];
-      return 'hsl(' + (hue[0] +(hue[1] - hue[0]) * Math.random()) + ',' +
+      return "hsl(" + (hue[0] +(hue[1] - hue[0]) * Math.random()) + ',' +
```
Comment (kgrgreer) [inline src/foam/color/colorlib.js]: Why this change? The coding standard is to use ' unless there's a good reason to use " or `.

File: src/foam/nanos/controller/ApplicationController.js [hunk 2]
```diff
@@ -752,7 +753,7 @@ foam.CLASS({
       }
 
       return new Promise(function(resolve, reject) {
-        self.stack.push(self.StackBlock.create({ view: { class: 'foam.u2.view.LoginView', mode_: 'SignIn' }, parent: self }));
+        self.stack.push(self.StackBlock.create({ view: { ...(self.loginVariables?.loginView ?? { class: 'foam.u2.view.LoginView' }), mode_: 'SignIn' }, parent: self }));
```
Comment (kgrgreer) [inline src/foam/nanos/controller/ApplicationController.js]: Why are we doing this? loginVariables seems like another context? Can't we just register a replacement for the LoginView?

File: src/foam/nanos/theme/Theme.js [hunk 3]
```diff
@@ -198,6 +198,11 @@ foam.CLASS({
       name: 'headConfig',
       writePermissionRequired: true
     },
+    {
```
Comment (kgrgreer) [inline src/foam/nanos/theme/Theme.js:201]: Didn't theme's used to have the ability to register class substitutions?

File: src/foam/nanos/controller/ApplicationController.js (hunk 2 above)
Comment (sarthak-marwaha) [reply src/foam/nanos/controller/ApplicationController.js]: loginView on theme can be a viewSpec so we need to pass the whole viewSpec not just the class

File: src/foam/nanos/theme/Theme.js (hunk 3 above)
Comment (sarthak-marwaha) [reply src/foam/nanos/theme/Theme.js:201]: It does, but this allows us to spec other props on the view as well

File: src/foam/nanos/u2/navigation/SignIn.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ foam.CLASS({
     'ctrl',
     'currentMenu',
     'loginSuccess',
+    'loginVariables',
```
Comment (kgrgreer) [inline src/foam/nanos/u2/navigation/SignIn.js]: Where is this exported?

Comment (sarthak-marwaha) [reply src/foam/nanos/u2/navigation/SignIn.js]: Application Controller
